compiler/modules/front/codeGen.bal

```diff
@@ -359,11 +360,54 @@ function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialE
             env.assignments.push(...effect.assignments);
         }
     }                
+    check unusedLocalVariables(cx, env, initialEnv.bindings);
     int[] assignments = [];
     addAssignments(assignments, env.assignments, startRegister);
     return { block: curBlock, assignments };
 }
 
+function unusedLocalVariables(CodeGenContext cx, Environment env, Binding? bindingLimit) returns CodeGenError? {
+    Binding? tem = env.bindings;
+    while true {
```

Comment:
This looks too complicated. You can stop when `tem === bindingLimit`. If we haven't reached that, then tem must be non-nil.

Reply:
Fixed